Bug: Channel closed error when I fetch records from Web Console

### What happened?

If I fetch records via Web Console, I see the following errors in logs:

```
2025-03-29 07:12:43.855 (45248) [ERROR] -- storage/entry/io/record_reader.rs:207 Failed to send record stress_test/entry-1/1743231265267496: channel closed
2025-03-29 07:12:44.506 (39552) [ERROR] -- storage/entry/io/record_reader.rs:207 Failed to send record stress_test/entry-1/1743231267256090: channel closed
2025-03-29 07:12:46.298 (80064) [ERROR] -- storage/entry/io/record_reader.rs:207 Failed to send record stress_test/entry-1/1743231271257802: channel closed
2025-03-29 07:12:46.488 (68032) [ERROR] -- storage/entry/io/record_reader.rs:207 Failed to send record stress_test/entry-1/1743231277253541: channel closed
2025-03-29 07:12:46.527 (97152) [ERROR] -- storage/entry/io/record_reader.rs:207 Failed to send record stress_test/entry-1/1743231279238281: channel closed
2025-03-29 07:12:46.567 ( 2848) [ERROR] -- storage/entry/io/record_reader.rs:207 Failed to send record stress_test/entry-1/1743231281257548: channel closed
2025-03-29 07:12:46.607 ( 8544) [ERROR] -- storage/entry/io/record_reader.rs:207 Failed to send record stress_test/entry-1/1743231283252731: channel closed

```

### Version

1.14.4

### OS

Linux

### Snippet to repeat the bug

_No response_

### Logs

```Shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- RS-658: fix server-side channel error for empty conditional query, [PR-88](https://github.com/reductstore/web-console/pull/88)
+
 ### [1.9.0] - 2025-02-25
 
 ### Fixed

--- a/src/Views/BucketPanel/EntryDetail.test.tsx
+++ b/src/Views/BucketPanel/EntryDetail.test.tsx
@@ -144,6 +144,13 @@ describe("EntryDetail", () => {
     expect(rows.at(1).text()).toContain("2.0 KB");
     expect(rows.at(1).text()).toContain("text/plain");
     expect(rows.at(1).text()).toContain('"type": "test"');
+
+    expect(bucket.query).toBeCalledWith("testEntry", undefined, undefined, {
+      head: true,
+      limit: 10,
+      strict: true,
+      when: {},
+    });
   });
 
   it("should toggle between ISO and Unix timestamps", async () => {

--- a/src/Views/BucketPanel/EntryDetail.tsx
+++ b/src/Views/BucketPanel/EntryDetail.tsx
@@ -51,13 +51,20 @@ export default function EntryDetail(props: Readonly<Props>) {
     setIsLoading(true);
     setRecords([]);
     setWhenError("");
+
     try {
       const bucket = await props.client.getBucket(bucketName);
+
       const options = new QueryOptions();
       options.limit = limit;
       options.head = true;
       options.strict = true;
-      if (whenCondition.trim()) options.when = JSON.parse(whenCondition);
+      if (whenCondition.trim().length > 0) {
+        options.when = JSON.parse(whenCondition);
+      } else {
+        options.when = {}; // enable POST endpoint that handle the head flag correctly
+      }
+
       for await (const record of bucket.query(entryName, start, end, options)) {
         setRecords((records) => [...records, record]);
       }
@@ -234,6 +241,9 @@ export default function EntryDetail(props: Readonly<Props>) {
               matchBrackets: true,
               autoCloseBrackets: true,
             }}
+            onChange={(editor, data, value) => {
+              setWhenCondition(formatJSON(value));
+            }}
             onBeforeChange={(editor, data, value) => {
               setWhenCondition(value);
             }}
@@ -255,7 +265,11 @@ export default function EntryDetail(props: Readonly<Props>) {
           </Typography.Text>
         </div>
         <div className="fetchButton">
-          <Button onClick={handleFetchRecordsClick} type="primary">
+          <Button
+            onClick={handleFetchRecordsClick}
+            type="primary"
+            id="fetchButton"
+          >
             Fetch Records
           </Button>
         </div>

--- a/src/Views/BucketPanel/EntryDetail.tsx
+++ b/src/Views/BucketPanel/EntryDetail.tsx
@@ -26,6 +26,7 @@ import "codemirror/mode/javascript/javascript";
 
 // @ts-ignore
 import prettierBytes from "prettier-bytes";
+
 interface Props {
   client: Client;
   permissions?: TokenPermissions;
@@ -241,9 +242,6 @@ export default function EntryDetail(props: Readonly<Props>) {
               matchBrackets: true,
               autoCloseBrackets: true,
             }}
-            onChange={(editor, data, value) => {
-              setWhenCondition(formatJSON(value));
-            }}
             onBeforeChange={(editor, data, value) => {
               setWhenCondition(value);
             }}
@@ -265,11 +263,7 @@ export default function EntryDetail(props: Readonly<Props>) {
           </Typography.Text>
         </div>
         <div className="fetchButton">
-          <Button
-            onClick={handleFetchRecordsClick}
-            type="primary"
-            id="fetchButton"
-          >
+          <Button onClick={handleFetchRecordsClick} type="primary">
             Fetch Records
           </Button>
         </div>

--- a/src/Views/BucketPanel/EntryDetail.tsx
+++ b/src/Views/BucketPanel/EntryDetail.tsx
@@ -63,7 +63,7 @@ export default function EntryDetail(props: Readonly<Props>) {
       if (whenCondition.trim().length > 0) {
         options.when = JSON.parse(whenCondition);
       } else {
-        options.when = {}; // enable POST endpoint that handle the head flag correctly
+        options.when = {}; // enable POST endpoint that handles the head flag correctly
       }
 
       for await (const record of bucket.query(entryName, start, end, options)) {

